embassy-nrf/src/uarte.rs

```diff
@@ -275,6 +430,31 @@ impl<'d, T: Instance> Write for Uarte<'d, T> {
     }
 }
 
+impl<'a, T: Instance> Drop for UarteRx<'a, T> {
+    fn drop(&mut self) {
+        info!("uarte rx drop");
+
+        let r = T::regs();
+
+        let did_stoprx = r.events_rxstarted.read().bits() != 0;
+        info!("did_stoprx {}", did_stoprx);
+
+        // Wait for rxto, if needed.
+        while did_stoprx && r.events_rxto.read().bits() == 0 {}
+
+        // Finally we can disable, and we do so for the peripheral
+        // i.e. not just rx concerns.
+        r.enable.write(|w| w.enable().disabled());
+
+        gpio::deconfigure_pin(r.psel.rxd.read().bits());
+        gpio::deconfigure_pin(r.psel.txd.read().bits());
+        gpio::deconfigure_pin(r.psel.rts.read().bits());
+        gpio::deconfigure_pin(r.psel.cts.read().bits());
+
+        info!("uarte drop: done");
```

Comment:
This bit of peripheral dropping is the only bit that made me think... is it the responsibility of the tx or the rx to do this? I'm not sure that it matters.